functions/spanner/src/test/java/functions/HelloSpannerTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package functions;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.cloud.functions.HttpRequest;
+import com.google.cloud.functions.HttpResponse;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.ErrorCode;
+import com.google.cloud.spanner.ReadContext;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.SpannerExceptionFactory;
+import com.google.cloud.spanner.Statement;
+import com.google.common.testing.TestLogHandler;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.logging.Logger;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+
+@RunWith(JUnit4.class)
+public class HelloSpannerTest {
+  @Mock private HttpRequest request;
+  @Mock private HttpResponse response;
+  @Mock private DatabaseClient client;
+
+  private static final Logger LOGGER = Logger.getLogger(HelloSpanner.class.getName());
+  private static final TestLogHandler logHandler = new TestLogHandler();
+  private static boolean originalUseParentHandlers;
+
+  private BufferedWriter writerOut;
+  private StringWriter responseOut;
+
+  @BeforeClass
+  public static void setupTestLogging() {
+    LOGGER.addHandler(logHandler);
+    originalUseParentHandlers = LOGGER.getUseParentHandlers();
+    LOGGER.setUseParentHandlers(false);
+  }
+
+  @AfterClass
+  public static void restoreLogging() {
+    LOGGER.removeHandler(logHandler);
+    LOGGER.setUseParentHandlers(originalUseParentHandlers);
+  }
+
+  @Before
+  public void beforeTest() throws IOException {
+    Mockito.mockitoSession().initMocks(this);
+
+    request = PowerMockito.mock(HttpRequest.class);
+    response = PowerMockito.mock(HttpResponse.class);
+    client = PowerMockito.mock(DatabaseClient.class);
+
+    BufferedReader reader = new BufferedReader(new StringReader("{}"));
```

Comment:
Nit: do you need these two lines?

Reply:
I'm ok w/ this in a test.  In the sample, I'd want to see this using GSON or JsonObject rather than as a constant.

Reply:
Sorry for not being clear - I think these two lines don't do anything useful.

(If you're copying this from one of my samples - I ended up removing them in quite a few of mine.)

Reply:
Yep, that was shameless copy-pasting, and it's not needed. Removed.